src/os/bluestore/BlueStore.cc

```diff
@@ -6028,21 +6028,25 @@ int BlueStore::_do_alloc_write(
 	csum_order = std::min(wctx->csum_order, ctz(l->length()));
       }
     }
-    while (final_length > 0) {
+
+    int count = 0;
+    std::vector<AllocExtent> extents = 
+                std::vector<AllocExtent>(final_length / min_alloc_size);
+
+    int r = alloc->alloc_extents(final_length, min_alloc_size, max_alloc_size,
+                                 hint, &extents, &count);
+    need = 0;
```

Comment:
added logic to update needed according to allocation in each write and then unreserve at end of function.